docs/content/how-to/kubevirt/troubleshooting-kubevirt-cluster.md

```diff
@@ -0,0 +1,63 @@
+# Troubleshooting a KubeVirt cluster
+
+## Fetching VM bootstrap logs
+
+The VM console logs are really useful to troubleshoot issues when the HyperShift/KubeVirt VM nodes are not correctly joining the cluster.
+KubeVirt v1.1.0 will expose logs from the serial console of guest VMs in a k8s native way (`kubectl logs -n <namespace> <vmi_pod> -c guest-console-log`) but his is still not available with KubeVirt v1.0.0.
+
+On KubeVirt v1.0.0 you can use a helper script to stream them from interactive console executed in background.
+
+```bash
+#!/bin/bash
+NAME=$1
+
+if [[ -z "${NAME}" ]]
+then
+    echo "Please specify the name of the guest cluster."
+    exit 1
+fi
+
+HSNS=clusters
+VMNS="${HSNS}"-"${NAME}"
+REPLICAS=$(oc get NodePool -n "${HSNS}" "${NAME}" -o=jsonpath='{.spec.replicas}')
+PLATFORMTYPE=$(oc get NodePool -n "${HSNS}" "${NAME}" -o=jsonpath='{.spec.platform.type}')
+INFRAID=$(oc get HostedCluster -n "${HSNS}" "${NAME}" -o=jsonpath='{.spec.infraID}')
+
+if [[ "${PLATFORMTYPE}" != "KubeVirt" ]]; then
+    echo "This tool is designed for the KubeVirt provider."
+    exit 1
+fi
+
+if [[ "${PLATFORMTYPE}" != "KubeVirt" ]]; then
+    echo "This tool is designed for the KubeVirt provider."
+    exit 1
+fi
+
+if ! which tmux >/dev/null 2>&1;
+then
+    echo "this tool requires tmux, please install it."
+    exit 1
+fi
+
+VMNAMES=()
+
+while [[ ${#VMNAMES[@]} < ${REPLICAS}  ]]; do
+  for VMNAME in $(oc get vmi -n "${VMNS}" -l hypershift.openshift.io/infra-id="${INFRAID}" -o name 2>/dev/null); do
+    SVMNAME=${VMNAME/virtualmachineinstance.kubevirt.io\//}
+    if ! [[ " ${VNMANES[*]} " =~ ${SVMNAME} ]]; then
+	   VMNAMES+=("${SVMNAME}")
+	   tmux new-session -s "${SVMNAME}" -d "virtctl console --timeout 30 -n ${VMNS} ${SVMNAME} | tee -a ${VMNS}_${SVMNAME}.log"
```

Comment:
what happens if the vmi is not running yet when this script executes `virtctl console`? will it eventually connect, or is it possible for it to timeout waiting for the vmi to come online?

Reply:
ah, makes sense!